static/app/views/settings/project/projectKeys/details/loaderSettings.tsx

```diff
@@ -0,0 +1,266 @@
+  },
+};
+
+export function LoaderSettings({keyId, organizationId, projectId, projectKey}: Props) {
+  const api = useApi();
+  const [browserSdkVersion, setBrowserSdkVersion] = useState(
+    projectKey.browserSdkVersion
+  );
+  const [dynamicSDKLoaderOptions, setDynamicSDKLoaderOptions] = useState(
+    projectKey.dynamicSdkLoaderOptions
+  );
+
+  useEffect(() => {
+    setBrowserSdkVersion(projectKey.browserSdkVersion);
+  }, [projectKey.browserSdkVersion]);
+
+  useEffect(() => {
+    setDynamicSDKLoaderOptions(projectKey.dynamicSdkLoaderOptions);
+  }, [projectKey.dynamicSdkLoaderOptions]);
+
+  const apiEndpoint = `/projects/${organizationId}/${projectId}/keys/${keyId}/`;
+  const loaderLink = getDynamicText({
+    value: projectKey.dsn.cdn,
+    fixed: '__JS_SDK_LOADER_URL__',
+  });
+
+  const handleToggleDynamicSDKLoaderOption = useCallback(
+    async <T extends typeof dynamicSDKLoaderOptions, K extends keyof T>(
+      dynamicSdkLoaderOption: K,
+      value: T[K]
+    ) => {
+      const newDynamicSdkLoaderOptions = Object.keys(dynamicSDKLoaderOptions).reduce(
+        (acc, key) => {
+          if (key === dynamicSdkLoaderOption) {
+            return {...acc, [key]: value};
+          }
+          return {...acc, [key]: dynamicSDKLoaderOptions[key]};
+        },
+        {}
+      );
+
+      addLoadingMessage();
+
+      try {
+        const response = await api.requestPromise(apiEndpoint, {
+          method: 'PUT',
+          data: {
+            dynamicSdkLoaderOptions: newDynamicSdkLoaderOptions,
+          },
+        });
+
+        setDynamicSDKLoaderOptions(response.dynamicSdkLoaderOptions);
+
+        addSuccessMessage(t('Successfully updated dynamic SDK loader configuration'));
+      } catch (error) {
+        const message = t('Unable to updated dynamic SDK loader configuration');
+        handleXhrErrorResponse(message)(error);
+        addErrorMessage(message);
+      }
+    },
+    [api, apiEndpoint, dynamicSDKLoaderOptions, setDynamicSDKLoaderOptions]
+  );
+
+  const handleUpdateBrowserSDKVersion = useCallback(
+    async (newBrowserSDKVersion: typeof browserSdkVersion) => {
+      addLoadingMessage();
+
+      const apiData: {
+        browserSdkVersion: typeof browserSdkVersion;
+        dynamicSdkLoaderOptions?: Partial<Record<DynamicSDKLoaderOption, boolean>>;
+      } = {
+        browserSdkVersion: newBrowserSDKVersion,
+      };
+
+      const shouldRestrictDynamicSdkLoaderOptions =
+        !sdkVersionSupportsPerformanceAndReplay(newBrowserSDKVersion);
+
+      if (shouldRestrictDynamicSdkLoaderOptions) {
+        // Performance & Replay are not supported before 7.x
+        const newDynamicSdkLoaderOptions = {
+          ...dynamicSDKLoaderOptions,
+          hasPerformance: false,
+          hasReplay: false,
+        };
+
+        apiData.dynamicSdkLoaderOptions = newDynamicSdkLoaderOptions;
+      }
+
+      try {
+        const response = await api.requestPromise(apiEndpoint, {
```

Comment:
it is fine to still use `api.requestPromise` but we have been replacing those with `useApiQuery`

Reply:
Yeah, I just moved this over from keySettings, but can also rewrite this :) Maybe I'll do this in a follow up, though!